PhysicsTools/Heppy/interface/MuScleFitCorrector.h

```diff
@@ -0,0 +1,312 @@
+    if( scaleParNum != scaleParVecSize ) {
+      std::cout << "Error: inconsistent number of parameters: scaleFunction has "<<scaleParNum<<" parameters but "<<scaleParVecSize<<" have been read from file" << std::endl;
+      exit(1);
+    }
+    
+    
+    std::vector<double>::const_iterator scaleParIt = scaleParVec_.begin();
+    std::vector<double>::const_iterator resolDataParIt = resolDataParVec_.begin();
+    std::vector<double>::const_iterator resolMCParIt = resolMCParVec_.begin();
+    
+    scaleParArray_ = new double[scaleParNum];
+    resolDataParArray_ = new double[resolParNum];
+    resolMCParArray_ = new double[resolParNum];
+    
+    for ( int iPar=0; iPar<scaleParNum; ++iPar) {
+      double parameter = *scaleParIt;
+      scaleParArray_[iPar] = parameter;
+      ++scaleParIt;
+    }
+    
+    if (useResol_){
+      for ( int iPar=0; iPar<resolParNum; ++iPar) {
+	double parameter = *resolDataParIt;
+	resolDataParArray_[iPar] = parameter;
+	++resolDataParIt;
+      }
+      
+      for ( int iPar=0; iPar<resolParNum; ++iPar) {
+	double parameter = *resolMCParIt;
+	resolMCParArray_[iPar] = parameter;
+	++resolMCParIt;
+      }
+    }
+    
+  }
+  //----------------------------//
+  
+  
+  // Parser of the parameters file
+  void readParameters(const TString& fileName){
+    scaleParArray_ = 0;
+    resolDataParArray_ = 0;
+    resolMCParArray_ = 0;
+    
+    // Read the parameters file
+    ifstream parametersFile(fileName.Data());
+    
+    if( !parametersFile.is_open() ) {
+      std::cout << "Error: file " << fileName << " not found. Aborting." << std::endl;
+      abort();
+    }
+    
+    std::string line;
+    int nReadPar = 0;
+    std::string iteration("Iteration ");
+    // Loop on the file lines
+    while (parametersFile) {
+      
+      getline( parametersFile, line );
+      size_t lineInt = line.find("value");
+      size_t iterationSubStr = line.find(iteration);
+      
+      if( iterationSubStr != std::string::npos ) {
+	
+	std::stringstream sLine(line);
+	std::string num;
+	int scaleFunctionNum = 0;
+	int resolutionFunctionNum = 0;
+	int wordCounter = 0;
+	
+	// Warning: this strongly depends on the parameters file structure.                                                                                   
+	while( sLine >> num ) {
+	  ++wordCounter;
+	  if( wordCounter == 8 ) {
+	  std::stringstream in(num);
+	  in >> resolutionFunctionNum;
+	  }
+	  
+	  if( wordCounter == 9 ) {
+	    std::stringstream in(num);
+	    in >> scaleFunctionNum;
+	  }
+	}
+	
+	scaleFunctionId_ = scaleFunctionNum;
+	scaleFunction_ = scaleFunctionService(scaleFunctionNum);
+	resolutionFunctionId_ = resolutionFunctionNum;
+	resolutionFunction_ = resolutionFunctionService(resolutionFunctionNum);
+	
+	/*       std::cout<<"Function IDs: "<<std::endl; */
+	/*       std::cout<<"     scale function number "<<scaleFunctionId_<<std::endl; */
+	/*       std::cout<<"     resolution function number "<<resolutionFunctionId_<<std::endl; */
```

Comment:
why should people remove commented code that could be useful for debugging??? 
comments, including stuff that can help debugging should be welcome unless they completly clutter the rest of the code (not the case here)